### ISSUE TITLE
Feature Request: API call to list tags

### ISSUE THREAD

**Issue by zopanix** [go-gitea/gitea#5693]
### Feature Request
It would be nice to have a REST API call for listing tags on a repository.

It would mean adding a new API call:

#### Description
Type: GET
Path: /repos/{owner}/{repo}/tags

I would think the response could be as simple as:
```
[
  "String",
  "String"
]
```
Because the call assumes you know for which {owner} and {repo} you're making the call for.

**Comment by zopanix** [comment]
Just for tracking purposes, here's what github proposes as response object:
Type: `GET`
Path: `/repos/{owner}/{repo}/tags`

```
[
  {
    "name": "v0.1",
    "commit": {
      "sha": "c5b97d5ae6c19d5c5df71a34c7fbeeda2479ccbc",
      "url": "https://api.github.com/repos/octocat/Hello-World/commits/c5b97d5ae6c19d5c5df71a34c7fbeeda2479ccbc"
    },
    "zipball_url": "https://github.com/octocat/Hello-World/zipball/v0.1",
    "tarball_url": "https://github.com/octocat/Hello-World/tarball/v0.1"
  }
]
```
I personally have no opinion on what the response "should" be, but I can imagine following what gihub does will make it easier to integrate gitea with other tools.

Please post something if you start working on this, I might be able to get some time on this as well, altough my golang experience is only about 2 weeks, I'd love to help but will need some guidance.

**Comment by lafriks** [comment]
You can already get list of tags with:
```
GET /repos/{owner}/{repo}/git/refs/tags
```
For example: https://try.gitea.io/api/v1/repos/lafriks/vscode/git/refs/tags

**Comment by lunny** [comment]
closed by #5850
